packages/studio-plugin/src/parsers/helpers/UtilityTypeParsingHelper.ts

```diff
@@ -99,4 +101,39 @@ export default class UtilityTypeParsingHelper {
       type: reducedShape,
     };
   };
+
+  private static handleRequired: UtilityTypeHandler = (
+    typeArgs,
+    parseTypeNode
+  ) => {
+    if (typeArgs.length !== 1) {
+      throw new Error(
+        `One type param expected for Required utility type. Found ${typeArgs.length}.`
+      );
+    }
+
+    const originalType = parseTypeNode(typeArgs[0]);
+
+    if (originalType.kind !== ParsedTypeKind.Object) {
+      return originalType;
```

Comment:
Is it valid TypeScript to have Required applied to a Type that's not an Object (for example a `string` or `string` union)? If that isn't valid TypeScript, I'd say we shouldn't handle gracefully here.

Reply:
Something like `Required<string>` is valid TypeScript. Based on my testing, it doesn't give any errors and it matches the original type. Out of the utility types I've implemented, the only one with unpredictable behavior for non-Object types was `Pick`, which is why I opted to throw an error there